src/test/regress/specs/isolation_replace_wait_function.spec

```diff
@@ -0,0 +1,69 @@
+# allow isolation tester to recognize locks on workers
+
+session "s1"
+
+step "pg-96-rename"
+{
+  -- break this into a separate step because each step can be at most 1024 chars
+  ALTER FUNCTION pg_catalog.pg_blocking_pids(integer)
+    RENAME TO old_pg_blocking_pids;
+}
+
+step "pg-96-create"
+{
+  CREATE FUNCTION pg_catalog.pg_blocking_pids(pTarget integer)
+  RETURNS int4[] AS $$
+    DECLARE
+      mLocalPids int4[];
+      mRemotePids int4[];
+      mTransactionNum int8;
+    BEGIN
+      SELECT pg_catalog.old_pg_blocking_pids(pTarget) INTO mLocalPids;
+
+      IF (array_length(mLocalPids, 1) > 0) THEN
+        RETURN mLocalPids;
+      END IF;
+
+      SELECT transaction_number INTO mTransactionNum FROM get_all_active_transactions() WHERE process_id = pTarget;
+      
+      SELECT array_agg(process_id) INTO mRemotePids FROM (
+        WITH map AS (
+          SELECT process_id, transaction_number FROM get_all_active_transactions()
+        ), edge AS (
+          SELECT blocking_transaction_num FROM dump_global_wait_edges()
+          WHERE waiting_transaction_num = mTransactionNum
+        )
+        SELECT map.process_id FROM map, edge
+        WHERE map.transaction_number = edge.blocking_transaction_num
+      ) AS sub;
+
+      RETURN mRemotePids;
+    END;
+  $$ LANGUAGE plpgsql;
+}
+
+step "pg10-setup"
+{
+  ALTER FUNCTION pg_catalog.pg_isolation_test_session_is_blocked(integer, integer[])
+    RENAME TO old_pg_isolation_test_session_is_blocked;
+  
+  CREATE FUNCTION pg_catalog.pg_isolation_test_session_is_blocked(pTarget integer, pSources integer[])
+  RETURNS boolean AS $$
+    DECLARE
+      mTransactionNum int8;
+    BEGIN
+      IF pg_catalog.old_pg_isolation_test_session_is_blocked(pTarget, pSources) THEN
+        RETURN true;
+      END IF;
+
+      SELECT transaction_number INTO mTransactionNum FROM get_all_active_transactions() WHERE process_id = pTarget;
+
+      -- pg doesn't think we're blocked locally, so check whether we're blocked globally
+      RETURN EXISTS (
+        SELECT 1 FROM dump_global_wait_edges() WHERE waiting_transaction_num = mTransactionNum
+      );
+    END;
+  $$ LANGUAGE plpgsql;
+}
+
```

Comment:
Wonder if it would be better to maintain the code for this in the extension, and just redirect the function here. That way we'd avoid the length limits and maintaining the expected/alternate files.